zio-http/src/main/scala/zio/http/model/headers/values/Authorization.scala

```diff
@@ -0,0 +1,147 @@
+package zio.http.model.headers.values
+
+import zio.http.model.headers.values.Authorization.AuthScheme.{Basic, Bearer, Digest, Unparsed}
+
+import java.net.URI
+import java.nio.charset.StandardCharsets
+import java.util.Base64
+import scala.collection.immutable.HashMap
+import scala.util.Try
+
+/** Authorization header value. */
+sealed trait Authorization
+
+object Authorization {
+
+  sealed trait AuthScheme
+
+  object AuthScheme {
+    final case class Basic(username: String, password: String)            extends AuthScheme
+    final case class Digest(
+      response: String,
+      username: String,
+      realm: String,
+      uri: URI,
+      opaque: String,
+      algorithm: String,
+      qop: String,
+      cnonce: String,
+      nc: Int,
+      userhash: Boolean,
+      rest: Map[String, String],
```

Comment:
IMO, we should rename that to something else than `rest` as it may misleading (especially can be similar to REST).

Reply:
I removed it altogether. I thought I would need it but it's not necessary.